alpha/template/template.go

```diff
@@ -0,0 +1,115 @@
+package template
+
+import (
+	"context"
+	"io"
+	"slices"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+)
+
+// BundleRenderer defines the function signature for rendering bundle images
+type BundleRenderer func(context.Context, string) (*declcfg.DeclarativeConfig, error)
+
+// Template defines the common interface for all template types
+type Template interface {
+	// RenderBundle renders a bundle image reference into a DeclarativeConfig
+	RenderBundle(ctx context.Context, image string) (*declcfg.DeclarativeConfig, error)
+	// Render processes the template input and returns a DeclarativeConfig
+	Render(ctx context.Context, reader io.Reader) (*declcfg.DeclarativeConfig, error)
+	// Schema returns the schema identifier for this template type
+	Schema() string
+}
+
+// TemplateFactory creates template instances based on schema
+type TemplateFactory interface {
```

Comment:
this is template package, could we shorter the name to `Factory`, so that we have `template.Factory` references in the code?